src/components/duration.rs

```diff
@@ -400,6 +434,18 @@ impl Duration {
         }
     }
 
+    /// Returns the result of adding a `Duration` to the current `Duration`
+    #[inline]
+    pub fn add(&self, other: &Self) -> TemporalResult<Self> {
+        self.internal_add(other)
```

Comment:
Nit: Dldo we need to have an `internal_add` method? I think it would be cleaner to move the code of `internal_add` to `add`, then just call `self.add(other.negated())` for `subtract`.